.github/workflows/install-valkey/action.yml

```diff
@@ -0,0 +1,97 @@
+name: Install Valkey
+
+inputs:
+    engine-version:
+        description: "Engine version to install"
+        required: true
+        type: string
+    target:
+        description: "Specified target toolchain, ex. x86_64-unknown-linux-gnu"
+        type: string
+        required: true
+        options:
+            - x86_64-unknown-linux-gnu
+            - aarch64-unknown-linux-gnu
+            - x86_64-apple-darwin
+            - aarch64-apple-darwin
+            - aarch64-unknown-linux-musl
+            - x86_64-unknown-linux-musl
+
+env:
+    CARGO_TERM_COLOR: always
+    VALKEY_MIN_VERSION: "7.2.5"
+
+runs:
+    using: "composite"
+
+    steps:
+        - name: Cache Valkey
+           # TODO: remove the musl ARM64 limitation when https://github.com/actions/runner/issues/801 is resolved
+          if: ${{ inputs.target != 'aarch64-unknown-linux-musl' }}
+          uses: actions/cache@v4
+          id: cache-valkey
+          with:
+            path: |
+                ~/valkey
+            key: valkey-${{ inputs.engine-version }}-${{ inputs.target }}
+
+        - name: Build Valkey
+          if: ${{ steps.cache-valkey.outputs.cache-hit != 'true' }}
+          shell: bash
+          run: | 
+            echo "Building valkey ${{ inputs.engine-version }}"
+            cd ~
+            git clone https://github.com/valkey-io/valkey.git
+            cd valkey
+            git checkout ${{ inputs.engine-version }}
+            make BUILD_TLS=yes
+
+        - name: Install Valkey
+          shell: bash
+          run: |
+            cd ~/valkey
+            if command -v sudo &> /dev/null
+            then
+                echo "sudo command exists"
+                sudo make install
+            else
+                echo "sudo command does not exist"
+                make install
+            fi
+            echo 'export PATH=/usr/local/bin:$PATH' >>~/.bash_profile
+
+        - name: Check if the current version is a Valkey release
+          id: is_valkey_release
+          uses: ./.github/workflows/compare-versions
+          with:
+              version1: ${{ inputs.engine-version }}
+              version2: ${{ env.VALKEY_MIN_VERSION }}
+
+        - name: Verify Valkey installation and symlinks
+          shell: bash
+          run: | 
+            IS_VALKEY=${{ steps.is_valkey_release.outputs.result }}
```

Comment:
You explained it well it the PR description, but there's a lot of bash code here which will be hard to understand without the description (in the future when it is not PR). Please add docs - why we need symlink if we using valkey source code, what each bunch of lines doe's, etc.

Reply:
added